backend/routes/user.js

```diff
@@ -0,0 +1,20 @@
+import { Router } from 'express';
+import { adminHandler, authenticationHandler } from '../middlewares/auth-middleware.js';
+import {
+  changeUserRoleHandler,
+  deleteUserHandler,
+  getAllUserHandler,
+} from '../controllers/user-controller.js';
+
+const router = Router();
+
+// get all users
+router.get('/', authenticationHandler, adminHandler, getAllUserHandler);
+
```

Comment:
Can we see what can be the best naming conventions? here the middlewares and the actual router handler are getting clubbed and a bit confused, lets see if we have some standard practice

Reply:
I checked both auth-middleware file and post-controller file for reference. Same naming convention is also used there.

I think we can drop the "handler" at the end of each function's name.
"getAllUserHandler" --> "getAllUser" 

and we can add "check" in the beginning of the middleware functions
"authenticationHandler" ---> "checkAuthenticate"

Let me know what you think

Reply:
Based on my research I felt the following more easy to understand, can we follow this, let me know if you have the simpler appraoch. Because in a long run "check" might get confusing ig.

<img width="647" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/33affb3c-37f1-4ab7-ac5c-ebbfba7541d1">
<img width="680" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/9da53f03-c434-4ba3-b2ce-a6818af9fd7f">

Reply:
okay 
Let's go with this. 
It gives more clarity
I will change the names accordingly.